models/balancer/ethereum/balancer_vebal_slopes.sql

```diff
@@ -0,0 +1,129 @@
+{{
+    config(
+        schema = 'balancer_ethereum',
+        alias='vebal_slopes',
+        post_hook='{{ expose_spells(\'["ethereum"]\',
+                                    "project",
+                                    "balancer",
+                                    \'["markusbkoch", "mendesfabio", "stefenon"]\') }}'
+    ) 
+}}
+
+
+WITH base_locks AS (
+        SELECT d.provider, ts AS locked_at, locktime AS unlocked_at, ts AS updated_at
+        FROM {{ source('balancer_ethereum', 'veBAL_call_create_lock') }} l
+        JOIN {{ source('balancer_ethereum', 'veBAL_evt_Deposit') }} d
+        ON d.evt_tx_hash = l.call_tx_hash
+        
+        UNION ALL
+        
+        SELECT provider, cast(null as numeric(38)) AS locked_at, locktime AS unlocked_at, ts AS updated_at
+        FROM {{ source('balancer_ethereum', 'veBAL_evt_Deposit') }}
+        WHERE value = 0
+    ),
+    
+    decorated_locks AS (
+        SELECT
+          provider, unlocked_at, updated_at, FIRST_VALUE(locked_at) OVER (PARTITION BY provider, locked_partition ORDER BY updated_at) AS locked_at
+        FROM (
+          SELECT
+            *,
+            SUM(CASE WHEN locked_at IS NULL THEN 0 ELSE 1 END) OVER (PARTITION BY provider ORDER BY updated_at) AS locked_partition
+          FROM base_locks
+        ) AS foo
+    ),
+    
+    locks_info AS (
+        SELECT *, unlocked_at - locked_at AS lock_period
+        FROM decorated_locks
+    ),
+
+    deposits AS (
+        SELECT
+            provider,
+            evt_block_number AS block_number,
+            evt_block_time AS block_time,
+            SUM(value/1e18) AS delta_bpt
+        FROM {{ source('balancer_ethereum', 'veBAL_evt_Deposit') }}
+        GROUP BY 1, 2, 3
+    ),
+    
+    withdrawals AS (
+        SELECT 
+            provider,
+            evt_block_number AS block_number,
+            evt_block_time AS block_time,
+            -SUM(value/1e18) AS delta_bpt
+        FROM {{ source('balancer_ethereum', 'veBAL_evt_Withdraw') }}
+        GROUP BY 1, 2, 3
+    ),
```

Comment:
Do you think we can apply incremental filters to CTEs for withdrawal events and deposit events(and maybe other CTEs if possible)?

Reply:
Done!